chat_downloader/sites/youtube.py

```diff
@@ -418,8 +418,7 @@ def __init__(self, **kwargs):
     _YT_INITIAL_BOUNDARY_RE = r'\s*(?:var\s+meta|</script|\n)'
     _YT_INITIAL_DATA_RE = r'(?:window\s*\[\s*["\']ytInitialData["\']\s*\]|ytInitialData)\s*=\s*({.+?})\s*;' + \
         _YT_INITIAL_BOUNDARY_RE
-    _YT_INITIAL_PLAYER_RESPONSE_RE = r'ytInitialPlayerResponse\s*=\s*({.+?})\s*;' + \
-        _YT_INITIAL_BOUNDARY_RE
+    _YT_INITIAL_PLAYER_RESPONSE_RE = r"""(?:^\s*window\["ytInitialPlayerResponse"\]|var ytInitialPlayerResponse) = (\{.+?\});(?:\s*$|</script>|var )"""
```

Comment:
Thanks for the fix and I can confirm that it works, but I think it would be better if you could just change `_YT_INITIAL_BOUNDARY_RE` to `\s*(?:var\s+(?:meta|head)|</script|\n)` on line 418:

```diff
-    _YT_INITIAL_BOUNDARY_RE = r'\s*(?:var\s+meta|</script|\n)'
+    _YT_INITIAL_BOUNDARY_RE = r'\s*(?:var\s+(?:meta|head)|</script|\n)'
```

Or just remove the `meta` in it (like what you did) to fix it once and for all (probably):
```diff
-    _YT_INITIAL_BOUNDARY_RE = r'\s*(?:var\s+meta|</script|\n)'
+    _YT_INITIAL_BOUNDARY_RE = r'\s*(?:var\s+|</script|\n)'
```

This way the existing code was reused and the regex is much simpler and easy to understand.

I've tested both fixes above with the videos in the original issue #145 and they worked for both of them:

- https://www.youtube.com/watch?v=1cEi9NanicE (provided by ArthurKun021)
- https://www.youtube.com/watch?v=7tFgcf0YcEA (provided by you)

which I am not a member of, but the `ytInitialPlayerResponse` variable still exists. So I also tested a video which I have member access to: https://www.youtube.com/watch?v=S8qYn4lC8ew, and the fixes above also worked.

P.S. YouTube also changed the response for unlisted live videos, and I can also confirm that this fixes the issue.

Reply:
Hi, Thank you for the review.
Yeah, you right. `_YT_INITIAL_BOUNDARY_RE = \s*(?:var\s+(?:meta|head)|</script|\n)` can work and reused the existing code.